lib/pluginmanager/offline_plugin_packager.rb

```diff
@@ -0,0 +1,120 @@
+# encoding: utf-8
+require "pluginmanager/ui"
+require "pluginmanager/errors"
+require "bootstrap/environment"
+require "bootstrap/util/compress"
+require "paquet"
+require "stud/temporary"
+require "fileutils"
+
+module LogStash module PluginManager
+  class SpecificationHelpers
+    WILDCARD = "*"
+    WILDCARD_INTO_RE = ".*"
+
+    def self.find_by_name_with_wildcards(pattern)
+      re = transform_pattern_into_re(pattern)
+      ::Gem::Specification.find_all.select do |specification|
+        specification.name =~ re
+      end
+    end
+
+    def self.transform_pattern_into_re(pattern)
+      Regexp.new(/^#{pattern.gsub(WILDCARD, WILDCARD_INTO_RE)}$/)
+    end
+  end
+
+  class OfflinePluginPackager
+    LOGSTASH_DIR = "logstash"
+    DEPENDENCIES_DIR = ::File.join(LOGSTASH_DIR, "dependencies")
+
+    # To make sure we have the maximun compatibility
+    # we will ignore theses gems and they wont be included in the pack
+    IGNORE_GEMS_IN_PACK = %w(
+      logstash-core
+      logstash-core-queue-jruby
+      logstash-core-event-java
+      logstash-core-plugin-api
+      jar-dependencies
+    )
+
+    INVALID_PLUGINS_TO_EXPLICIT_PACK = IGNORE_GEMS_IN_PACK.collect { |name| Regexp.new(/^#{name}/) } + [
+      /mixin/
```

Comment:
so you can't pack gems if they depend on mixins? or the pack will be broken because it doesn't include the mixin?